assets/app/view/game/route_selector.rb

```diff
@@ -258,31 +276,70 @@ def actions(render_halts)
           store(:routes, @routes)
         end
 
+        add_train = lambda do
+          @step.add_train(@routes)
+          store(:routes, @routes)
+        end
+
+        remove_train = lambda do
+          @step.remove_train(@selected_route)
+          store(:routes, @routes)
+        end
+
+        increase_train = lambda do
+          @step.increase_train(@selected_route)
+          store(:routes, @routes)
+        end
+
+        decrease_train = lambda do
+          @step.decrease_train(@selected_route)
+          store(:routes, @routes)
+        end
+
         submit_style = {
           minWidth: '6.5rem',
           marginTop: '1rem',
           padding: '0.2rem 0.5rem',
         }
 
-        revenue = begin
-          @game.format_revenue_currency(@game.routes_revenue(active_routes))
-        rescue Engine::GameError
-          '(Invalid Route)'
-        end
+        revenue = if @step.respond_to?(:total_str)
+                    begin
+                      @step.total_str(@game.routes_revenue(active_routes))
+                    rescue Engine::GameError
+                      @step.revenue_fail
+                    end
+                  else
+                    begin
+                      "Submit #{@game.format_revenue_currency(@game.routes_revenue(active_routes))}"
+                    rescue Engine::GameError
+                      '(Invalid Route)'
+                    end
+                  end
 
         render_halts ||= @game.respond_to?(:routes_subsidy) && @game.routes_subsidy(active_routes).positive?
         subsidy = render_halts ? " + #{@game.format_currency(@game.routes_subsidy(active_routes))} (subsidy)" : ''
-        buttons = [
-          h('button.small', { on: { click: clear } }, 'Clear Train'),
-          h('button.small', { on: { click: clear_all } }, 'Clear All'),
-          h('button.small', { on: { click: reset_all } }, 'Reset'),
-        ]
-        if @game_data.dig('settings', 'auto_routing') || @game_data['mode'] == :hotseat
+
+        show_ct_button = !@step.respond_to?(:show_clear_train_button?) || @step.show_clear_train_button?
```

Comment:
why!?

Reply:
Removing buttons because it's confusing for track conversion. I will build a new view/action.